content/cloud-docs/run/ui.mdx

```diff
@@ -63,6 +63,15 @@ If you would rather automatically apply plans that don't have errors, you can [e
 
 To help you review proposed changes, Terraform Cloud can automatically run [speculative plans][speculative plan] for pull requests or merge requests.
 
+## Testing Terraform Upgrades with Speculative Plans
+
+Using the UI, you can start a new speculative plan with the workspace's current configuration version and any Terraform version available to the organization.
+
+1. Click **Start New Run** and see the **Start a new run** modal window.
+2. Select the **Plan only** option as the run type. This run mode reveals a new option to choose the specific Terraform version.
+3. Visit the workspace settings and change the [workspace terraform version](/cloud-docs/workspaces/settings#terraform-version) setting.
+4. Start a new run using the procedure above but this time select the type **Allow empty apply**. Creating a run in **Allow empty apply** mode allows Terraform to apply a plan containing no infrastructure changes. During an empty apply, Terraform upgrades the state file format to match the Terraform version, if required. This is particularly useful for upgrading between Terraform versions in your workspace. Once this run is applied, the state version upgrade will be complete.
```

Comment:
```suggestion
1. Click **Actions > Start New Run**. The **Start a new run** box appears.
1. Select **Plan only** as the run type. 
1. Select a version from the **Choose Terraform version** menu. The speculative plan will use this version without changing the workspace's settings.
1. Click **Start run**.

If the run is successful, you can change the workspace's Terraform version and [upgrade the state](/cloud-docs/workspaces/state#upgrading-state).
```

Reply:
Hmm, I think these instructions wandered off course a little bit, both in the current diff and Laura's suggestion. 

It looks like we're trying to outline TWO distinct but related procedures here: 

- Do a plan-only run with a different Terraform version to find out whether it'll work OK or not.
- Actually flip the workspace Terraform version...
    - ...and, _maybe,_ do an allow empty apply run to lock in a state version upgrade. (Only necessary on certain version transitions, most notably when landing on 0.13.)

I think the instructions will be more clear if they treat those two actions as distinct.

Reply:
I agree that it's two separate things! But I thought (let me know if I'm super off base here) that this edit really outlines the process for the first procedure, which is what this section is about. This is essentially how you test, right? You run the speculative (plan-only) run with the version you want selected and see if it's successful? And since this is called "Testing Terraform Upgrades with Speculative Plans", I was thinking it makes more sense to really only outline the procedure to do that.

And then my thought is that you can basically say "cool! your test was successful? Do you wanna actually change the version now and/or do the state migration? Here are links to that. This is what that last sentence was attempting to do.

That said, please let me know if I'm way off base here --> I just wanted to provide some context for this edit. It may be that there are gaps in my knowledge/that my execution of this was unclear. Anyway, I value your opinion so let me know what you think!

Reply:
@laurapacilio You know, I'm not sure where my head was at on Friday, but on re-reading your suggested text, I agree with your edit! Possibly I got confused about the state of the header. Comment retracted.

Reply:
okay! I still appreciate you and your feedback! :)